markdown/solace-k8s-operator/solace-k8s-operator.md

```diff
@@ -0,0 +1,214 @@
+author: Ken Barr
+summary: How to build a Solace PubSubPlus Kubernetes Operator
+id: solace-k8s-operator
+tags: iguide
+categories: Kubernetes
+environments: Web
+status: Draft
+feedback link:
+analytics account: UA-3921398-10
+
+# How to build a Solace PubSubPlus Kubernetes Operator
+
+## Overview
+The kubernetes [Operator Pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator) is exactly what it sounds like.  It is code or automation that acts like the set of runbooks a human operator of an application would use to deploy, monitor, troubleshoot, upgrade and decommission an application.  The operator can do these functions because it has a deep understanding of the application it is operating on and knowns if the application is functioning correctly. 
+
+When Solace first introduced a Kubernetes solution back in the days of Kubernetes 1.5/1.6, Operator concepts where very much in flux.  For this reason, we based our Solution on Helm to do initial deployments and rolling upgrades and config-maps as the glue that binds together the highly available cluster of brokers together. But today Kubernetes Operators are far more prevalent and stable.  They make extending Kubernetes behaviour to manage stateful applications like event brokers and databases possible via custom resource and custom resource definitions. So for organizations that manage there deployments through custom resources and operators a Solace operator would be usefull, here is why.  
+
+All Kubernetes deployments are a collection of resources. Services, Persistent Volumes, Secrets, Pods, are example of basic resources that are composed together to make up a deployment.  With a Solace custom resources as defined in an operator, then a Solace highly available cluster could also be used as another resource to compose a deployment.  Its life cycle tied to the deployment and maintained within the deployment.  It would be possible for example to upgrade brokers and applications in one operation.
+
+The easiest first step toward a Solace Operator is to wrap the existing Helm charts and Kubernetes templates into an operator based with the Operator SDK.  So, lets see how to do that.
+
+## Environment Setup
+* First to build operators you will need [golang](https://golang.org/doc/install) with compatible compiler.
+* [Next install the operator-sdk](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/install-operator-sdk.md) which can be done as a simple binary download or built locally.
+* Finally configure the helm values.yaml to suite your target environment.  In this example I am installing a Highly Available cluster in minikube which would be good for development testing including failure and retry conditions:
+    - cd ~/git_repos; 
```

Comment:
This block could have a code format with ```

Reply:
code formatted the cli commands, not the instructional steps.